common/src/main/java/com/wynntils/features/utilities/TranscribeMessagesFeature.java

```diff
@@ -0,0 +1,173 @@
+import com.wynntils.core.text.StyledText;
+import com.wynntils.core.text.StyledTextPart;
+import com.wynntils.handlers.chat.event.ChatMessageReceivedEvent;
+import com.wynntils.handlers.chat.event.NpcDialogEvent;
+import com.wynntils.handlers.chat.type.NpcDialogueType;
+import com.wynntils.models.wynnalphabet.WynnAlphabet;
+import com.wynntils.utils.colors.ColorChatFormatting;
+import com.wynntils.utils.type.IterationDecision;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.Style;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@ConfigCategory(Category.UTILITIES)
+public class TranscribeMessagesFeature extends Feature {
+    @Persisted
+    public final Config<Boolean> transcribeChat = new Config<>(true);
+
+    @Persisted
+    public final Config<Boolean> transcribeNpcs = new Config<>(true);
+
+    @Persisted
+    public final Config<TranscribeCondition> transcribeCondition = new Config<>(TranscribeCondition.ALWAYS);
+
+    @Persisted
+    public final Config<Boolean> coloredTranscriptions = new Config<>(true);
+
+    @Persisted
+    public final Config<ColorChatFormatting> gavellianColor = new Config<>(ColorChatFormatting.LIGHT_PURPLE);
+
+    @Persisted
+    public final Config<ColorChatFormatting> wynnicColor = new Config<>(ColorChatFormatting.DARK_GREEN);
+
+    private static final Pattern END_OF_HEADER_PATTERN = Pattern.compile(".*[\\]:]\\s?");
+    private static final Pattern WYNNIC_NUMBER_PATTERN = Pattern.compile("[⑴-⑿]+");
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public void onChat(ChatMessageReceivedEvent event) {
+        if (!transcribeChat.get()) return;
+        if (!Models.WynnAlphabet.hasWynnicOrGavellian(event.getStyledText().getString())) return;
+
+        boolean transcribeWynnic = Models.WynnAlphabet.shouldTranscribe(transcribeCondition.get(), WynnAlphabet.WYNNIC);
+        boolean transcribeGavellian =
+                Models.WynnAlphabet.shouldTranscribe(transcribeCondition.get(), WynnAlphabet.GAVELLIAN);
+
+        if (!transcribeWynnic && !transcribeGavellian) return;
+
+        StyledText styledText = event.getStyledText();
+
+        StyledText modified = getStyledTextWithTranscription(styledText, transcribeWynnic, transcribeGavellian, false);
+
+        if (styledText.getString().equalsIgnoreCase(modified.getString())) return;
+
+        event.setMessage(modified.getComponent());
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public void onNpcDialogue(NpcDialogEvent event) {
+        if (!transcribeNpcs.get()) return;
+        if (!Models.WynnAlphabet.hasWynnicOrGavellian(event.getChatMessage().toString())) return;
+
+        boolean transcribeWynnic = Models.WynnAlphabet.shouldTranscribe(transcribeCondition.get(), WynnAlphabet.WYNNIC);
```

Comment:
Since shouldTranscribe() is in the model, the condition enum needs to move there as well. Please put it as a separate file under the "type" subpackage.